### ISSUE TITLE
FlxSprite.stamp() not working on WebGL

### ISSUE THREAD

**Issue by ttencate** [HaxeFlixel/flixel#1987]
- **Flixel version:** 4.2.0
- **OpenFL version:** 3.6.1
- **Lime version:** 2.9.1
- **Affected targets:** webgl (with or without `-Dnext`)

---

Using `stamp()` seems to make the target sprite unusable. It just stops showing anything. Whether it's broken or just fully transparent is hard to tell.

I narrowed it down to two lines in the `stamp` method, version A below. If I simplify it even further, to version B, something shows up, but it doesn't blend against the original sprite; it just overwrites its pixels and alpha channel.

Expected result (using canvas, or any other backend):

![2016-10-28-121100_1357x1175_scrot](https://cloud.githubusercontent.com/assets/90930/19803275/1057e34a-9d08-11e6-91cb-ad0a487a8f80.png)

Version A (not showing anything):

![2016-10-28-121118_1357x1175_scrot](https://cloud.githubusercontent.com/assets/90930/19803281/1e7176bc-9d08-11e6-952a-e0ba340542a3.png)

Version B (not blending against the red target sprite):

![2016-10-28-121153_1357x1175_scrot](https://cloud.githubusercontent.com/assets/90930/19803288/28ed76f4-9d08-11e6-9158-1829208ef9f1.png)

**Code snippet reproducing the issue**:

``` haxe
package;

import flixel.FlxG;
import flixel.FlxSprite;
import flixel.FlxState;
import flixel.util.FlxColor;

class MenuState extends FlxState
{
    override public function create():Void
    {
        super.create();

        // Works on webgl, no effect on canvas. For canvas we use Project.xml.
        FlxG.camera.bgColor = FlxColor.LIME;

        var cards = new FlxSprite();
        cards.loadGraphic(AssetPaths.cards__png);

        var sprite = new FlxSprite();
        sprite.makeGraphic(cards.graphic.width, cards.graphic.height, FlxColor.RED);

        // Version A (what FlxSprite.stamp() does): transparent (or no?) image on webgl
        cards.drawFrame();
        sprite.graphic.bitmap.draw(cards.framePixels);

        // Version B: no blending on webgl, just overwrites alpha channel
        // sprite.graphic.bitmap.draw(cards.graphic.bitmap);

        add(sprite);
    }
}
```

You need `cards.png`; any PNG file with transparency will do.

---

I _suspect_ this is due to a bug in OpenFL 3.6.1, which is not present in OpenFL 4.0.0. Namely, when I call `getPixel32()` on the `BitmapData` before drawing it onto another, I get the same effect in WebGL: nothing shows up. But without the `getPixel32()` call, the bitmap appears just fine. My suspicion is that `getPixel32` somehow gets the `BitmapData` into a broken state, and that HaxeFlixel is doing something similar whereby it ends up in the same broken state.

``` haxe
import openfl.display.Bitmap;
import openfl.display.BitmapData;
import openfl.display.Sprite;
import openfl.Assets;

class Main extends Sprite {
    public function new () {
        super ();

        var openflBitmap = Assets.getBitmapData("assets/openfl.png");

        // Comment out this line, and the bitmap shows up in webgl too:
        openflBitmap.getPixel32(0, 0);

        var bitmapData = new BitmapData(openflBitmap.width, openflBitmap.height);
        bitmapData.draw(openflBitmap);

        addChild(new Bitmap(bitmapData));
    }
}
```

I considered reporting this to OpenFL, but (as far as I know) the 3.x version is no longer maintained, so there would be little point.

Because it seems HaxeFlixel will be stuck with OpenFL 3 at least for now, it would be great if this could be worked around. Especially because I suspect it's also the cause of https://github.com/HaxeFlixel/flixel-ui/issues/197 in flixel-ui.
##

**Comment by ttencate** [comment]
As a workaround, I've found that `source.frame.paint(dest, new openfl.geom.Point(x, y))` is a working substitute for `dest.stamp(source, x, y)` in my case.

**Comment by Gama11** [comment]
It looks like this is now working correctly with OpenFL 8 (see #2136), wich makes sense since you mentioned this was already fixed way back in OpenFL 4.0.0.
